azurelinuxagent/common/utils/archive.py

```diff
@@ -48,99 +48,19 @@
     re.compile(r"waagent_status\.(\d+)\.json$")
 ]
 
-_GOAL_STATE_PATTERN = re.compile(r"^(.*)/GoalState\.(\d+)\.xml$", re.IGNORECASE)
-
-# Old names didn't have incarnation, new ones do. Ensure the regex captures both cases.
+# Old names have incarnation, new ones don't. Ensure the regex captures both cases.
 # 2018-04-06T08:21:37.142697_incarnation_N
 # 2018-04-06T08:21:37.142697_incarnation_N.zip
 _ARCHIVE_PATTERNS_DIRECTORY = re.compile(r"^\d{4}\-\d{2}\-\d{2}T\d{2}:\d{2}:\d{2}\.\d+(_incarnation_(\d+))?$$")
 _ARCHIVE_PATTERNS_ZIP = re.compile(r"^\d{4}\-\d{2}\-\d{2}T\d{2}:\d{2}:\d{2}\.\d+(_incarnation_(\d+))?\.zip$")
 
-
```

Comment:
Now GoalState saves the goal state directly to the history folder, so there is not need to flush it